### PR TITLE
Add spiffe-csi driver support to authentication part

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

/cc @pxp928 @mlieberman85 @sudo-bmitch @wlynch @dentrax

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
Add spiffe-csi driver support to authenticaiotn part
```

### CODE REVIEW

**Review by wlynch — APPROVED** [review]
/lgtm

Thanks!

**Review by pxp928 — APPROVED** [review]
LGTM

File: docs/authentication.md [hunk 1]
```diff
@@ -161,3 +161,21 @@ to deployment `volumes` section:
         hostPath:
           path: /run/spire/sockets/agent.sock
 ```
+
+Last but not least, thanks to [spiffe-csi](https://github.com/spiffe/spiffe-csi), which is a a CSI (Container Storage Interface) driver for Kubernetes that facilitates injection of the SPIFFE Workload API , there is alternative way of retrieving the Agent socket to your Pods without having to mount the hostPath. You can read more about it [here](https://www.kusari.dev/blog/spiffee-csi/). Once you have installed `spiffe-csi-driver` into your cluster by following the [installation steps](https://github.com/spiffe/spiffe-csi/tree/main/example#steps) in the GitHub repository, the only thing that you have to do is add the following code snippet to deployment `volumes` and `volumeMounts` sections:
```
Comment (Dentrax) [inline docs/authentication.md:165]:
```diff
- which is a a CSI
+ which is a CSI
```

```diff
- hostPath
+ `hostPath`
```

**Comment by lcarva** [comment]
@developer-guy, a rebase should fix the CI test.

**Comment by developer-guy** [comment]
I'm going to do that ASAP 🫡

**Comment by developer-guy** [comment]
It seems that everything is fine now.
/cc @wlynch @lcarva

**Review by lcarva — APPROVED** [review]
/lgtm

**Pull request merged**